src/WebJobs.Script.WebHost/Diagnostics/LinuxContainerEventGenerator.cs

```diff
@@ -66,6 +73,24 @@ private void ConsoleWriter(string evt)
             }
         }
 
+        private void TryReadEnvironmentVariables()
+        {
+            if (string.IsNullOrEmpty(_stampName))
+            {
+                _stampName = _environment.GetEnvironmentVariable("WEBSITE_HOME_STAMPNAME")?.ToLowerInvariant();
+            }
+
+            if (string.IsNullOrEmpty(_containerName))
```

Comment:
You can cache the containername in constructor it wont change ever and will be part of initial settings.